### ISSUE TITLE
Is there any way to create like broadcast server?

### ISSUE THREAD

**Issue by turusudiro** [otavepto/gbe_fork#55]
Is there a way to set up a broadcast server so that when I want to play multiplayer with my friends, I don't have to open the game and forward ports, especially since my friends are behind CGNAT? This would allow them to play with each other without me.

Currently, I am port forwarding, and my friends are using custom broadcasts with my public IP. Thanks!.

**Comment by turusudiro** [comment]
My current setup is vpn server with L2 on vps, so it can send broadcast. But i want to a simpler solution like run Goldberg Server to receive broadcast/packet (?) on vps so the client who wants to play together behind cgnat is possible to create a connection between each other.

**Comment by SeasonPhan** [comment]
i think we will need something like a master server to manage and connect all clients together?

**Comment by turusudiro** [comment]
> i think we will need something like a master server to manage and connect all clients together?

yeah i mean like this, it kinda sucks if other people on cgnat want to play together without a person who had a Public IP. Zerotier and Tailscale is a big no for us lol because the latency is crazy since my friends behind strict nat so its kinda impossible to hole punching NAT. Maybe we need some Master server that handle packets to clients that use custom_broadcast and set it to Master server?

**Comment by M4RCK5** [comment]
> > i think we will need something like a master server to manage and connect all clients together?
> 
> yeah i mean like this, it kinda sucks if other people on cgnat want to play together without a person who had a Public IP. Zerotier and Tailscale is a big no for us lol because the latency is crazy since my friends behind strict nat so its kinda impossible to hole punching NAT. Maybe we need some Master server that handle packets to clients that use custom_broadcast and set it to Master server?

What about [Lanemu](https://gitlab.com/Monsterovich/lanemu), its an decentralized vpn, you need at least one of u guys to open ports so that client can work as an relay server too.

Here is how to set it up:
![Lanemu](https://github.com/otavepto/gbe_fork/assets/20927212/a1532f79-a0e7-425a-9655-42d07ed9ecbf)

**Comment by Vlxst** [comment]
> > > i think we will need something like a master server to manage and connect all clients together?
> > 
> > 
> > yeah i mean like this, it kinda sucks if other people on cgnat want to play together without a person who had a Public IP. Zerotier and Tailscale is a big no for us lol because the latency is crazy since my friends behind strict nat so its kinda impossible to hole punching NAT. Maybe we need some Master server that handle packets to clients that use custom_broadcast and set it to Master server?
> 
> What about [Lanemu](https://gitlab.com/Monsterovich/lanemu), its an decentralized vpn, you need at least one of u guys to open ports so that client can work as an relay server too.
> 
> Here is how to set it up: ![Lanemu](https://private-user-images.githubusercontent.com/20927212/318196627-a1532f79-a0e7-425a-9655-42d07ed9ecbf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTI0NDM3MjgsIm5iZiI6MTcxMjQ0MzQyOCwicGF0aCI6Ii8yMDkyNzIxMi8zMTgxOTY2MjctYTE1MzJmNzktYTBlNy00MjVhLTk2NTUtNDJkMDdlZDllY2JmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDA2VDIyNDM0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWRlYmRiMDVmZWM1MTM4ZjZlYzUxNTNkY2I5ZWNkODI3MjI0Zjg2OGJmNTdhYzE4ZDUzZjM5NzJiMjNjMTA2MzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.MU8N_IekkRpqKjMwtFKcEqw1sRTG3iEJmKEiEWZBFTs)

Interesting, I'd like to see how this actually behaves

**Comment by LuKeSt0rm** [comment]
Could be interesting for the case: https://github.com/otavepto/gbe_fork/issues/233

**Comment by elyashivhazan** [comment]
> > i think we will need something like a master server to manage and connect all clients together?
> 
> yeah i mean like this, it kinda sucks if other people on cgnat want to play together without a person who had a Public IP. Zerotier and Tailscale is a big no for us lol because the latency is crazy since my friends behind strict nat so its kinda impossible to hole punching NAT. Maybe we need some Master server that handle packets to clients that use custom_broadcast and set it to Master server?

if ur friends under cgnat then they cant talk ptp they must go under a central server (what u suggested or at least what i understand from that suggestion) and ur setup is already doing it with zt1/tailscale. considering ur the one who set this up when ur offline ur friends can still play. this is what my group of friends r doing. i setup zt1 and they connect, im not even using `custom_broadcasts` because in windows (any os) u can just set the if's mtu to lets say 1 and all broadcast go to that vpn (considering the vpn supports L2 and zt1 dose)

**Comment by qwertysmosh** [comment]
> > > i think we will need something like a master server to manage and connect all clients together?
> > 
> > 
> > yeah i mean like this, it kinda sucks if other people on cgnat want to play together without a person who had a Public IP. Zerotier and Tailscale is a big no for us lol because the latency is crazy since my friends behind strict nat so its kinda impossible to hole punching NAT. Maybe we need some Master server that handle packets to clients that use custom_broadcast and set it to Master server?
> 
> What about [Lanemu](https://gitlab.com/Monsterovich/lanemu), its an decentralized vpn, you need at least one of u guys to open ports so that client can work as an relay server too.
> 
> Here is how to set it up: ![Lanemu](https://private-user-images.githubusercontent.com/20927212/318196627-a1532f79-a0e7-425a-9655-42d07ed9ecbf.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTg2MjAzNjIsIm5iZiI6MTcxODYyMDA2MiwicGF0aCI6Ii8yMDkyNzIxMi8zMTgxOTY2MjctYTE1MzJmNzktYTBlNy00MjVhLTk2NTUtNDJkMDdlZDllY2JmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjE3VDEwMjc0MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI0ZGYxZmM5NDNjZjYyMTM2NjI1YzRiM2I3YWI3NTcxODk1MTQyZjJjM2ViZjE3NmNmZDRiMDdiYmQxMmI1ZGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jmVF3rt4pkGrlrcPhy1qfJonky_Ip-1ITBCobcspnrQ)

If my network is behind CGNAT and I want to run it through playit.gg tunneling, what should I put into Known Hosts? Do I put 127.0.0.1:port, localhost:port or the IP Address provided by playit.gg?
